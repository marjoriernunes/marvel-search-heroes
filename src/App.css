.App {
  text-align: center;
}

.App-header {
  background-color: #FFF;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #000;
}

.App-link {
  color: #61dafb;
}

