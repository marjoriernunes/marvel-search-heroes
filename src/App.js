import react from 'react';
import './App.css';
import Header from './components/header/Header';
import SearchBar from './components/searchbar/SearchBar';
import Footer from './components/footer/Footer';
import CardList from './components/card/CardList';
import CharactersApi from './services/characters.service';
import Home from './pages/Home/Home';
import Icon from './assets/Icons';
import MarvelLogo from './assets/logo.svg';
import ToggleButton from './components/toggle/Toggle';

function App() {
  const [charactersInitial, charactersResponse] = react.useState(null);
  const [searchElement, setSearchElement] = react.useState('');
  const [orderChoosed, setOrder] = react.useState('name');

  const orderByName = (state) => {
    if (!!state) {
      setOrder('name');
    } else {
      setOrder('-name')
    }
  }

  react.useEffect(() => {
  
    const charactersRequest = async () => { 
      const data = await CharactersApi(searchElement, orderChoosed)
      charactersResponse(data)
    }
    charactersRequest();
  
  }, [searchElement, orderChoosed]);

  return (
      <div className="App">
        <img className="App-logo" src={MarvelLogo} alt="Marvel Search Heroes"/>
        <Header/>
        <SearchBar dataToFilter={charactersInitial?.heroes} sendData={(value) => {setSearchElement(value)}}/>
        <section className="options-bar">
          <p className="horoes-found">{`Encontramos ${charactersInitial?.heroes.length} ${charactersInitial?.heroes.length === 1 ? 'herói' : 'heróis'}`}</p>
          <div>
            <Icon name="ic_heroi" width="18" height="27"/>
            <span>Ordernar por nome - A/Z</span>
            <ToggleButton onChange={state => orderByName(state)} defaultChecked={true} />
          </div>
          <button>Somente favoritos</button>
        </section>
        <Home>
          <CardList itens={charactersInitial?.heroes}/>
        </Home>
        <Footer />
      </div>
  );
}

export default App;
