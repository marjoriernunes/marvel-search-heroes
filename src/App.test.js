import { render } from '@testing-library/react';
import App from './App';

test('renders correctly', () =>{
    const utils = render(<App/>)
    expect(utils.container).toMatchSnapshot()
});
