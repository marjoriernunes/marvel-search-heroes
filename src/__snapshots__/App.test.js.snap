// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      class="title"
    >
      explore o universo
    </h1>
    <p
      class="subtitle"
    >
      Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!
    </p>
    <div>
      <input
        placeholder="Pesquisaê"
        type="text"
      />
    </div>
    <div>
      <p>
        Encontramos undefined heróis
      </p>
      <button>
        Ordernar por nome - A/Z
      </button>
      <button>
        Somente favoritos
      </button>
    </div>
    <div>
      <ul
        class="container"
      />
    </div>
    <div
      class="footer-bar"
    />
  </div>
</div>
`;
